src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_rk3_step.jl

```diff
@@ -1,15 +1,15 @@
-using Oceananigans.Fields: location
+using Oceananigans.Fields: location, instantiated_location
 using Oceananigans.TurbulenceClosures: implicit_step!
 using Oceananigans.ImmersedBoundaries: retrieve_interior_active_cells_map, retrieve_surface_active_cells_map
 
-import Oceananigans.TimeSteppers: split_rk3_substep!, _split_rk3_substep_field!
+import Oceananigans.TimeSteppers: split_rk3_substep!, _split_rk3_substep_field!, store_fields!
```

Comment:
what does "store fields" mean?

Reply:
it's the same concept as `store_tendencies!` of the other RK3, however in this formulation of RK3 instead of storing the tendencies we store prognostic fields

Reply:
maybe `cache_previous_tendencies` and `cache_previous_fields` would be more clear. `store_tendencies` made sense because tendencies are only used for one thing (updating fields). But fields are used for many things, so "storing" fields could be used for a wide variety of purposes, like output, etc.

Reply:
sounds good! I ll open a new PR that does this change